File: CareForTheOld/Common/Helpers/ApiResponse.cs
namespace CareForTheOld.Common.Helpers;

/// <summary>
/// 统一 API 响应封装
/// </summary>
public class ApiResponse<T>
{
    public bool Success { get; set; }
    public T? Data { get; set; }
    public string Message { get; set; } = string.Empty;
    public List<string>? Errors { get; set; }

    public static ApiResponse<T> Ok(T data, string message = "操作成功")
        => new() { Success = true, Data = data, Message = message };

    public static ApiResponse<T> Fail(string message, List<string>? errors = null)
        => new() { Success = false, Message = message, Errors = errors };
}

/// <summary>
/// 分页数据封装
/// </summary>
public class PagedResult<T>
{
    public List<T> Items { get; set; } = [];
    public int TotalCount { get; set; }
    public int PageIndex { get; set; }
    public int PageSize { get; set; }
}

File: flutter_client/lib/core/models/api_response.dart
/// API 统一响应模型
class ApiResponse<T> {
  final bool success;
  final T? data;
  final String message;
  final List<String>? errors;

  const ApiResponse({
    required this.success,
    this.data,
    required this.message,
    this.errors,
  });

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(Map<String, dynamic>)? fromJsonT,
  ) {
    return ApiResponse(
      success: json['success'] as bool,
      data: json['data'] != null && fromJsonT != null
          ? fromJsonT(json['data'] as Map<String, dynamic>)
          : json['data'] as T?,
      message: json['message'] as String,
      errors: (json['errors'] as List<dynamic>?)?.cast<String>(),
    );
  }
}